src/GitHub.Exports/Services/VSServices.cs

```diff
@@ -68,6 +73,50 @@ public void ActivityLogWarning(string message)
             }
         }
 
+        /// <summary>Open a repository in Team Explorer</summary>
+        /// <remarks>
+        /// There doesn't appear to be a command that directly opens a target repo.
+        /// Our workaround is to create, open and delete a solution in the repo directory.
+        /// This triggers an event that causes the target repo to open. ;)
+        /// </remarks>
+        /// <param name="repoPath">The path to the repository to open</param>
+        /// <returns>True if a transient solution was successfully created in target directory (which should trigger opening of repository).</returns>
+        public bool TryOpenRepository(string repoPath)
+        {
+            bool solutionCreated = false;
+
+            try
+            {
+                var dte = (DTE)serviceProvider.GetService(typeof(DTE));
```

Comment:
Use `TryGetService<DTE>` for these calls, they're guaranteed not to throw. If it returns null, bail.